core/code/layerchooser.js

```diff
@@ -12,16 +12,39 @@
  */
 
 var LayerChooser = L.Control.Layers.extend({
+  options: {
+    // @option sortLayers: Boolean = true
+    // Ensures stable sort order (based on initial), while still providing ability to
+    // force specific order with layer's `sortPriority` option,
+    // which lower value means layer's upper position.
+    //
+    // If `sortPriority` is not specified - it will be set implicitly in increasing manner.
+    sortLayers: true,
+
+    sortFunction: function (layerA, layerB) {
+      var a = layerA.options.sortPriority || Infinity;
+      var b = layerB.options.sortPriority || Infinity;
+      return a < b ? -1 : (b < a ? 1 : 0);
+    }
+  },
+
   initialize: function (baseLayers, overlays, options) {
     this._overlayStatus = {};
     window.overlayStatus = this._overlayStatus; // compatibility
     this._mapToAdd = options && options.map;
     this.lastBaseLayerName = localStorage['iitc-base-map'];
+    this._lastPriority = -1000; // initial layers get priority <0
     L.Control.Layers.prototype.initialize.apply(this, arguments);
+    this._lastPriority = 0; // any following gets >0
   },
 
   _addLayer: function (layer, name, overlay) {
     L.Control.Layers.prototype._addLayer.apply(this, arguments);
+    // provide stable sort order
+    if (!layer.options.sortPriority) {
```

Comment:
technically, `0` may be a correct value

Reply:
`if (!('sortPriority' in layer.options)) {`

Reply:
or `!== undefined`
I use this one, but yours is more expressive :+1: